src/common/meta/src/wal/kafka/topic_selector.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+
+use serde::{Deserialize, Serialize};
+
+use crate::wal::kafka::topic_manager::Topic;
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+pub enum TopicSelectorType {
+    RoundRobin,
+}
+
+pub trait TopicSelector: Send + Sync {
+    fn select(&self, topic_pool: &[Topic]) -> Topic;
+}
+
+pub type TopicSelectorRef = Arc<dyn TopicSelector>;
+
+#[derive(Default)]
+pub struct RoundRobinTopicSelector {
+    cursor: AtomicUsize,
```

Comment:
Nit: do we need to persist the cursor value?

Reply:
Consider shuffle topics in the topic pool upon start.